src/react-queries/queries/thornode.ts

```diff
@@ -1,81 +1,106 @@
+        },
+      }
+    },
+  })
 }
 
-// Feature-agnostic, abstracts away THORNode endpoints
-export const thornode = createQueryKeys('thornode', {
-  poolData: (assetId: AssetId | undefined) => ({
-    queryKey: ['thornodePoolData', assetId],
-    queryFn: async () => {
-      if (!assetId) throw new Error('assetId is required')
+export const thornode = createThornodeQueries(
+  'thornode',
+  `${getConfig().VITE_THORCHAIN_NODE_URL}/thorchain`,
+)
 
-      const poolAssetId = assetIdToPoolAssetId({ assetId })
-      const { data } = await axios.get<ThornodePoolResponse>(
-        `${thornodeUrl}/thorchain/pool/${poolAssetId}`,
-      )
+export const mayanode = createThornodeQueries(
+  'mayanode',
+  `${getConfig().VITE_MAYACHAIN_NODE_URL}/mayachain`,
+)
 
-      return data
-    },
-  }),
-  poolsData: () => ({
-    queryKey: ['thornodePoolsData'],
-    queryFn: async () => {
-      const poolResponse = await thorService.get<ThornodePoolResponse[]>(
-        `${thornodeUrl}/thorchain/pools`,
-      )
+export const getInboundAddressesQuery = (chainId: ChainId) => {
+  const { queryKey, queryFn } = (() => {
+    switch (chainId) {
+      case thorchainChainId:
+        return thornode.inboundAddresses()
+      case mayachainChainId:
+        return mayanode.inboundAddresses()
+      default:
+        throw new Error(`Unsupported chainId: ${chainId}`)
+    }
+  })()
 
-      if (poolResponse.isOk()) {
-        return poolResponse.unwrap().data
-      }
+  return {
+    queryKey,
+    queryFn: queryFn as QueryFunction<InboundAddressResponse[]>,
+  }
+}
 
-      return []
-    },
-  }),
-  mimir: () => {
-    return {
-      queryKey: ['thorchainMimir'],
-      queryFn: fetchThorchainMimir,
+export const getMimirQuery = (chainId: ChainId) => {
+  const { queryKey, queryFn } = (() => {
+    switch (chainId) {
+      case thorchainChainId:
+        return thornode.mimir()
+      case mayachainChainId:
+        return mayanode.mimir()
+      default:
+        throw new Error(`Unsupported chainId: ${chainId}`)
     }
-  },
-  block: () => {
-    return {
-      queryKey: ['thorchainBlockHeight'],
-      queryFn: async () => {
-        const { data } = await axios.get<ThorchainBlock>(`${thornodeUrl}/thorchain/block`)
-        return data
-      },
-    }
-  },
-  inboundAddresses: () => {
-    return {
-      queryKey: ['thorchainInboundAddress'],
-      queryFn: async () => {
-        return (
-          // Get all inbound addresses
-          (
-            await thorService.get<InboundAddressResponse[]>(
-              `${thornodeUrl}/thorchain/inbound_addresses`,
-            )
-          ).andThen(({ data: inboundAddresses }) => {
-            // Exclude halted
-            const activeInboundAddresses = inboundAddresses.filter(a => !a.halted)
-            return Ok(activeInboundAddresses)
-          })
-        )
-      },
-    }
-  },
-})
+  })()
+
+  return {
+    queryKey,
+    queryFn: queryFn as QueryFunction<ThorchainMimir>,
+  }
+}
```

Comment:
This was the most ergonomic I could manage to allow for easiest possible conditional calls against thorchain or mayachain given a `chainId`. Open to feedback though.